server/helpers/cache.js

```diff
@@ -0,0 +1,38 @@
+const TIMEOUT = 24 * 60 * 60 * 1000 // 1 day
+const PRUNE_RATIO = 0.1
+
+class Cache {
+  constructor(limit, maxAge = TIMEOUT) {
+    this.limit = limit
+    this.maxAge = maxAge
+    this.cache = {}
+    this.size = 0
+  }
+
+  get(key) {
+    if (key in this.cache && this.cache[key].timestamp + this.maxAge > Date.now()) {
+      return this.cache[key].data
+    }
+    return undefined
+  }
+
+  set(key, value) {
+    if (this.get(key) === undefined) this.size++
+    this.cache[key] = {
+      timestamp: Date.now(),
+      data: value,
+    }
+    if (this.size > this.limit) this.prune()
+  }
+
+  prune() {
+    const pruneSize = Math.round(PRUNE_RATIO * this.limit)
```

Comment:
why not use a native Map? https://stackoverflow.com/questions/18541940/map-vs-object-in-javascript

A Map preserves the order of insertion, so you wouldn't even have to sort the keys, though arguably, last access time to the key is more relevant than the insertion time for the sake of cache pruning

Reply:
I wasn't aware that map preserves the order, refactored.

Reply:
I'd still leave here a comment clarifying that the order of insertion is preserved so an unaware reader isn't left wondering based on what do we actually prune the keys

Reply:
Fixed